### PR TITLE
Add implementation and use cases for the hal class

### PR DESCRIPTION
**Pull request by xujian009**

### CODE REVIEW

**Comment by tiannian** [comment]
Please solve CI's Error.

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+fn main() {}
```
Comment (tiannian) [inline src/main.rs]: main.rs? This crate's type is `lib`.

File: src/point.rs [hunk 2]
```diff
@@ -1,45 +1,99 @@
-use dislog_hal::DisLogPoint;
+use core::ops::{Add, Mul, Neg};
```
Comment (tiannian) [inline src/point.rs]: You can remove these traits import

File: src/scalar.rs [hunk 3]
```diff
@@ -1,46 +1,110 @@
...
 }
 
 impl ScalarInner {
-    fn new(scalar: curve25519_dalek::scalar::Scalar) -> Self {
+    pub fn new(scalar: [u8; 32]) -> Self {
+        Self {
+            data: curve25519_dalek::scalar::Scalar::from_bytes_mod_order(scalar),
+        }
+    }
+    pub(crate) fn setInner(scalar: curve25519_dalek::scalar::Scalar) -> Self {
```
Comment (tiannian) [inline src/scalar.rs]: Please according to the naming convention, use underline to name methods.

Comment (xujian009) [reply src/scalar.rs]: The modification has been submitted, all is fine

File: src/point.rs (hunk 2 above)
Comment (xujian009) [reply src/point.rs]: The modification has been submitted, all is fine

**Pull request merged**